multimodal/setup.py

```diff
@@ -64,7 +64,7 @@
     "onnx>=1.13.0,<1.14.0",
     "onnxruntime>=1.15.0,<1.16.0;platform_system=='Darwin'",
     "onnxruntime-gpu>=1.15.0,<1.16.0;platform_system!='Darwin'",
-    "tensorrt>=8.5.3.1,<8.5.4;platform_system=='Linux'",
+    "tensorrt>=8.5.3.1,<8.5.4;platform_system=='Linux' and python_version<'3.11'",  # tensorrt > 8.5.4 cause segfault
```

Comment:
@yinweisu could you give tensorrt 8.6.1 another shot? to get some more info about the segfault?

if i find some time, I'll try to install it in our py 3.11 staging cluster and try out `optimize_for_inference`

Reply:
Hi @ddelange, I've been trying to work out that problem with tensorrt 8.6.1 running the tests in `test_deployment_onnx.py`. It still gives me a segfault, I'll update with more info.

Reply:
Update: I ran some more tests in `python3.11`, and `tensorrt==8.6.1` worked with `cuda12.1` with `autogluon` built from source (current master) and `torch==2.1.1`. I was able to successfully run tests in `test_deployment_onnx.py` with the above versions.

All in all, we can wait for the cuda version to be upgraded inside autogluon requirements to cuda12.1 (https://github.com/autogluon/autogluon/pull/3663), which will then solve the segfault issue on the latest version of `tensorrt` and `cuda11.8`. Only then we can go ahead and update `tensorrt` version in setup.py

Reply:
Hi @AnirudhDagar :wave:

#3982 was merged. could you finalize support for `optimize_for_inference` on python 3.11 to finally close #2687?

Reply:
Yes, I'll update the issues as soon as we get rid of the py3.11 conditional which is still present there due to a few failing HPO tests ([see comment](https://github.com/autogluon/autogluon/pull/3982/files#r1526574710)). I need to run some tests locally, specifically for `optimize_for_inference`.

Reply:
@AnirudhDagar I saw you added "close https://github.com/autogluon/autogluon/issues/2687" to the release tracker https://github.com/autogluon/autogluon/issues/4027

but that issue tracks `optimize_for_inference` support on 3.11. is your plan to squeeze it in still?

Reply:
Thanks for the reminder @ddelange, yes I can give the `optimize_for_inference ` testing a shot after merging your ray PR.

Reply:
fwiw in a future version, the `tensorrt` python package seems deprecated (from inspecting the latest prerelease it's now an empty meta package forwarding to https://pypi.org/project/tensorrt-cu12/10.0.0b6/).

So if you hit a snatch, it might be worth to give `tensorrt-cu12` a try. See also the install instructions on the pypi page (to avoid an ugly pip-in-pip pattern ref https://github.com/NVIDIA/TensorRT/pull/3080)